### ISSUE TITLE
Support inheritance

### ISSUE THREAD

**Issue by RomkeVdMeulen** [MeirionHughes/aurelia-template-lint#167]
Not very common, but sometimes I have something like:

```ts
export class MyWidget extends Toggle {
  myProp = "Lorem Ipsum";
}
```

```ts
export class Toggle {
  toggled = false;

  toggle() {
    this.toggled = !this.toggled;
  }
}
```

with in `my-widget.html`:

```html
<template>
  <button click.trigger="toggle()">Toggle me</button>
  <p if.bind="toggled">${myProp}</p>
</template>
```

In which case the linter should not report `WARNING: cannot find 'toggled' in type 'MyWidget' Ln 3 Col 15 my-widget.html`.

**Comment by MeirionHughes** [comment]
Am I right in thinking Toggle is a DOM class?

**Comment by RomkeVdMeulen** [comment]
No, it's another ViewModel. Well, a base class for ViewModels.

Op vr 17 feb. 2017 12:15 schreef Meirion Hughes <notifications@github.com>:

**Comment by MeirionHughes** [comment]
Its possible your import statement is the problem as inheritance should be supported:  

https://github.com/MeirionHughes/aurelia-template-lint/commit/7f941b9f05b8b3dd458d98f20594f11342883dc5

**Comment by RomkeVdMeulen** [comment]
That could be it: I'm not importing with a relative path. I have updated my original post to show how I import `Toggle`.

**Comment by MeirionHughes** [comment]
Yes, that's it. I think you'll need to use relative paths. Unless I'm mistaken "src/components/toggle" means `/components/toggle` from within `src` module. If this is what you're trying doing then support for modules at the moment is done via setting the typings path - not tested with node_modules/@types though.

**Comment by RomkeVdMeulen** [comment]
Using relative paths isn't a good option for my project. `src` isn't an external module in my case, it refers to code under the `src` dir in my codebase. I'm using the TypeScript [baseURL compiler option](https://www.typescriptlang.org/docs/handbook/module-resolution.html#base-url) to make this work.

In this case the module path I'm importing from matches one of the paths I passed to `reflectionOpts.sourceFileGlob` exactly. Like this:

```ts
config.reflectionOpts.sourceFileGlob = "src/**/*.ts";
```

I can understand that following complex module resolution logic is a bit out of scope for a template linter. But perhaps it might be a useful feature to try to resolve non-relative imports that match the `sourceFileGlob`?
